src/features/foods/components/FoodSort.tsx

```diff
@@ -0,0 +1,90 @@
+import { TriangleDownIcon } from '@/assets/images/icons';
+import {
+  Icon,
+  Label,
+  SelectionDrawerContent,
+  SelectionDrawerRadioGroup,
+  SelectionDrawerRadioGroupItem,
+  SelectionDrawerRoot,
+  SelectionDrawerTrigger,
+} from '@/components/ui';
+import { cn } from '@/lib/tailwind/utils';
+
+import { useState } from 'react';
+
+export const sortOptions = [
+  {
+    id: 'createdAt',
+    label: 'Created At (Newest → Oldest)',
+  },
+  {
+    id: 'name',
+    label: 'Name (A → Z)',
+  },
+  {
+    id: 'expiry',
+    label: 'Expiry (Oldest → Newest)',
+  },
+];
+
+import { Route } from 'next';
+import { usePathname, useRouter, useSearchParams } from 'next/navigation';
+import React from 'react';
+
+export const FoodSort = () => {
+  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+  const [selectedOption, setSelectedOption] = useState(sortOptions[0].id);
+
+  const onValueChange = (value: string) => {
+    setSelectedOption(value);
+  };
+
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const { replace } = useRouter();
+
+  const handleSort = (term: string) => {
+    const params = new URLSearchParams(searchParams);
+    if (term) {
+      params.set('sort', term);
+    } else {
+      params.delete('sort');
+    }
+    replace(`${pathname}?${params.toString()}` as Route);
+  };
+
+  return (
+    <>
+      <SelectionDrawerRoot open={isDrawerOpen} onOpenChange={setIsDrawerOpen}>
+        <SelectionDrawerTrigger asChild>
+          <button className="flex items-center gap-2">
```

Comment:
There is an unwanted gap, which does not exist in Figma:
<img width="250" alt="image" src="https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99148565/d97951ad-18d4-46d0-a27d-2efc1bb5b85d">

Reply:
This text must match the selected sort.

<img width="300" alt="image" src="https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99148565/3d11597e-0ec3-4f87-97ba-0aed46174eba">